tmva/sofie/inc/TMVA/ROperator_Equal.hxx

```diff
@@ -0,0 +1,144 @@
+            }
+         }
+         // Broadcast B to Y
+         if (broadcastX2) {
+            if (model.IsInitializedTensor(fNX2)) {
+               auto data = model.GetInitializedTensorData(fNX2);
+               std::shared_ptr<void> broadcastedData(
+                  UTILITY::UnidirectionalBroadcast<float>(static_cast<float *>(data.get()), fShapeX2, fShapeY),
+                  std::default_delete<float[]>());
+               // Update the data and the shape of B
+               model.UpdateInitializedTensor(fNX2, model.GetTensorType(fNX2), fShapeY, broadcastedData);
+               fShapeX2 = fShapeY;
+            } else {
+               // Add an intermediate tensor for broadcasting B
+               fNBroadcastedX2 = "Broadcasted" + fNX2;
+               model.AddIntermediateTensor(fNBroadcastedX2, model.GetTensorType(fNX2), fShapeY);
+            }
+         }
+      } else {
+         fShapeY = fShapeX1;
+      }
+      model.AddIntermediateTensor(fNY, ETensorType::BOOL , fShapeY);
+   }
+
+   std::string Generate(std::string OpName){
+      OpName = "op_" + OpName;
+
+     if (fShapeY.empty()) {
+         throw std::runtime_error("TMVA SOFIE Equal Op called to Generate without being initialized first");
+      }
+      std::stringstream out;
+      out << "\n//------ Equal\n";
+      size_t length = ConvertShapeToLength(fShapeY);
+      // Broadcast A if it's uninitialized
+      if (!fNBroadcastedX1.empty()) {
+         out << SP << "// Broadcasting uninitialized tensor " << fNX1 << "\n";
+         out << SP << "{\n";
+         out << SP << SP << "float* data = TMVA::Experimental::SOFIE::UTILITY::UnidirectionalBroadcast<float>(tensor_" << fNX1 << ", " << ConvertShapeToString(fShapeX1) << ", " << ConvertShapeToString(fShapeY) << ");\n";
+         out << SP << SP << "std::copy(data, data + " << length << ", tensor_" << fNBroadcastedX1 << ");\n";
+         out << SP << SP << "delete[] data;\n";
+         out << SP << "}\n";
+      }
+      // Broadcast B if it's uninitialized
+      if (!fNBroadcastedX2.empty()) {
+         out << SP << "// Broadcasting uninitialized tensor " << fNX2 << "\n";
+         out << SP << "{\n";
+         out << SP << SP << "float* data = TMVA::Experimental::SOFIE::UTILITY::UnidirectionalBroadcast<float>(tensor_" << fNX2 << ", " << ConvertShapeToString(fShapeX2) << ", " << ConvertShapeToString(fShapeY) << ");\n";
+         out << SP << SP << "std::copy(data, data + " << length << ", tensor_" << fNBroadcastedX2 << ");\n";
+         out << SP << SP << "delete[] data;\n";
+         out << SP << "}\n";
+      }
+      const std::string& nameX1 = fNBroadcastedX1.empty()? fNX1 : fNBroadcastedX1;
+      const std::string& nameX2 = fNBroadcastedX2.empty()? fNX2 : fNBroadcastedX2;
+     
+      out << SP << "for (size_t id = 0; id < " << length << " ; id++){\n";
+      out << SP << SP << "tensor_" << fNY << "[id] = (tensor_" << nameX1 << "[id] == tensor_" << nameX2 <<"[id]) ? true : false;\n";
```

Comment:
Will it be good if we keep this operator as a Logical operator and also include the operations for Greater, Less, LessEqual, etc. other logical operators? As they won't have a lot of differences but a lot of code similarities.

cc: @lmoneta